### ISSUE TITLE
Speed up tbi.run_nth_year_tax_calc_model

### ISSUE THREAD

**Issue by hdoupe** [open-source-economics/Tax-Calculator#1710]
In OpenSourcePolicyCenter/PolicyBrain#750, we found that adding the AGI tables increased the time for one run from 15 to 40 seconds on the EC2 machines. 

On my local machine, it now takes 90 seconds to run one year.  Of the 90 seconds, about 70 seconds is spent in the `tbi_utils.summary` function.  I would like to use this issue to brainstorm ideas for improving the performance of this function.  

One idea is to reduce the number of pandas `cut` and `groupby` function calls.  In `tbi_utils.summary`, there is this snippet of code:
```
    # create difference tables grouped by adec
    df2['iitax'] = df2['iitax_adec']
    summ['diff_itax_adec'] = \
        create_difference_table(df1, df2,
                                groupby='weighted_deciles',
                                income_measure='c00100',
                                tax_to_diff='iitax')

    df2['payrolltax'] = df2['payrolltax_adec']
    summ['diff_ptax_adec'] = \
        create_difference_table(df1, df2,
                                groupby='weighted_deciles',
                                income_measure='c00100',
                                tax_to_diff='payrolltax')

    df2['combined'] = df2['combined_adec']
    summ['diff_comb_adec'] = \
        create_difference_table(df1, df2,
                                groupby='weighted_deciles',
                                income_measure='c00100',
                                tax_to_diff='combined')
```

What if we changed `create_diff_table` so that `tax_to_diff` could be a list of variable names?  Then, we perform the same operations on this 2-D array as we did on the 1-D array.  This would reduce the number of `create_difference_table` calls from 12 to 4.

Does anyone have any other ideas?

cc @martinholmer @MattHJensen

**Comment by martinholmer** [comment]
If the cost of the AGI tables is so high, why not just delete them and simplify the TaxBrain static results display (by removing the buttons that let you choose between AGI and expanded income?  How many people are interested in AGI distribution tables?

**Comment by MattHJensen** [comment]
+1 for removing AGI tables.

**Comment by hdoupe** [comment]
Ok sounds good to me. I'll open up a PR.

**Comment by martinholmer** [comment]
@hdoupe, why don’t you fix TaxBrain and I’ll fix Tax-Calculator?

**Comment by hdoupe** [comment]
That's fine with me. Thanks.

**Comment by martinholmer** [comment]
Shall I completely remove the AGI tables or add a switch to the tbi function that turns the AGI tables on and off?  @hdoupe @MattHJensen

**Comment by hdoupe** [comment]
I think you can just get rid of them.  If we want to add them back, then we can dig them up from the history of this file.  It looks like there's a disabled button for them here:

![screen shot 2017-11-28 at 4 48 43 pm](https://user-images.githubusercontent.com/9206065/33345995-25ba315c-d45c-11e7-991e-c2aa02e2f37c.png)

But, I have no idea how those buttons work.  I just tried to remove it and haven't had any success.  I'm also not sure how to activate it.  That part of the front-end is a mystery.  Until we figure out how it works, I don't think there's any use in having the extra code laying around.

**Comment by martinholmer** [comment]
@hdoupe said:
> But, I have no idea how those buttons work. I just tried to remove it and haven't had any success. I'm also not sure how to activate it. That part of the front-end is a mystery. 

Check with @GoFroggyRun on this.  He removed the "Excel Download" buttons on the TaxBrain results page.

**Comment by martinholmer** [comment]
Merge of pull request #1724 resolves issue #1710.
